src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,15 +1,310 @@
 import { Component } from 'react';
+import classNames from 'classnames';
+import './NewMovie.scss';
 
-type Props = {};
-type State = {};
+type Props = {
+  onAdd: (movie: Movie) => void,
+};
+
+type State = {
+  title: string,
+  description: string,
+  imgUrl: string,
+  imdbUrl: string,
+  imdbId: string,
+  isErrorTitle: boolean,
+  isErrorImgUrl: boolean,
+  isErrorImdbUrl: boolean,
+  isErrorImdbId: boolean,
+  isDisabled: boolean,
+};
+
+const initialState: State = {
+  title: '',
+  description: '',
+  imgUrl: '',
+  imdbUrl: '',
+  imdbId: '',
+  isErrorTitle: false,
+  isErrorImgUrl: false,
+  isErrorImdbUrl: false,
+  isErrorImdbId: false,
+  isDisabled: true,
+};
+
+// eslint-disable-next-line max-len
+const regExp = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@.\w_]*)#?(?:[.!/\\\w]*))?)$/;
 
 export class NewMovie extends Component<Props, State> {
-  state: State = {};
+  state: State = { ...initialState };
+
+  resetForms = () => {
+    this.setState({ ...initialState });
+  };
+
+  checkTitle = () => {
+    if (this.state.title.trim() === '') {
+      this.setState({
+        isErrorTitle: true,
+        isDisabled: true,
+      });
+    } else {
+      this.setState({
+        isErrorTitle: false,
+        isDisabled: false,
+      });
+    }
+  };
+
+  checkImgURl = () => {
+    if (!regExp.test(this.state.imgUrl)) {
+      this.setState({
+        isErrorImgUrl: true,
+        isDisabled: true,
+      });
+    } else {
+      this.setState({
+        isErrorImgUrl: false,
+        isDisabled: false,
+      });
+    }
+  };
+
+  checkImdbUrl = () => {
+    if (!regExp.test(this.state.imdbUrl)) {
+      this.setState({
+        isErrorImdbUrl: true,
+        isDisabled: true,
+      });
+    } else {
+      this.setState({
+        isErrorImdbUrl: false,
+        isDisabled: false,
+      });
+    }
+  };
+
+  checkImdbId = () => {
+    if (this.state.imdbId.trim() === '') {
+      this.setState({
+        isErrorImdbId: true,
+        isDisabled: true,
+      });
+    } else {
+      this.setState({
+        isErrorImdbId: false,
+        isDisabled: false,
+      });
+    }
+  };
+
+  checkMovie = () => {
+    const {
+      title,
+      description,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    } = this.state;
+
+    if (!this.state.isErrorTitle
+      && !this.state.isErrorImgUrl
+      && !this.state.isErrorImdbUrl
+      && !this.state.isErrorImdbId) {
+      this.props.onAdd({
+        title: title.trim(),
+        description: description.trim(),
+        imgUrl,
+        imdbUrl,
+        imdbId: imdbId.trim(),
+      });
+      this.resetForms();
+    }
+  };
 
   render() {
     return (
-      <form>
-        Put the form here
+      <form
+        className="newMovie"
+        method="post"
+        name="form"
+        onSubmit={(event) => {
+          event.preventDefault();
+          this.checkMovie();
+        }}
```

Comment:
call preventDefault on checkMovie

```suggestion
        onSubmit={this.checkMovie}
```